src/main/java/uk/gov/companieshouse/model/Reason.java

```diff
@@ -26,23 +28,41 @@ public void setOther(OtherReason other) {
         this.other = other;
     }
 
+    public IllnessReason getIllnessReason() { return this.illnessReason; }
+
+    public void setIllnessReason(IllnessReason illnessReason) {this.illnessReason = illnessReason; }
+
+    public ReasonType getReasonType() {
+        if(getOther() == null ^ getIllnessReason() == null){
```

Comment:
I would recommend keeping validation logic in the validator. This is a bit difficult to follow and will get more complex as more reason types are added.

Reply:
I'd still rather see a not null annotation on this and the validator removed.

Reply:
We can't use a NotNull annotation because the logic is that `Appeal.Reason` must contain either an `Other` or `Illness` reason. `Other` and `Illness` reasons have no common fields and therefore are separate models within `Reason`. We would need a custom annotation. I'm happy with either approach as long as the code is readable and clean.

Reply:
However Appeal.Reason can never be empty, which is why this method was added, so that we could use field level validation on this method to check that only one of the fields is occupied.
That's why we chose to return null when both are present.